pymeasure/instruments/agilent/agilentB1500.py

```diff
@@ -1123,14 +1117,13 @@ def ramp_source(
         step size, each separated by a pause.
 
         :param str source_type: Source type (``'Voltage'`` or ``'Current'``)
-        :param target_output: Target output voltage or current
-        :type: target_output: float
+        :param float target_output: Target output voltage or current
         :param int irange: Output range index
         :param float, optional comp: Compliance, defaults to previous setting
-        :param CompliancePolarity comp_polarity: Compliance polairty, defaults to auto
+        :param CompliancePolarity, optional comp_polarity: Compliance polairty, defaults to auto
         :param int or str, optional comp_range: Compliance ranging type, defaults to auto
-        :param stepsize: Maximum size of steps
-        :param pause: Duration in seconds to wait between steps
+        :param float, optional stepsize: Maximum size of steps
+        :param float, optional pause: Duration in seconds to wait between steps
```

Comment:
I don't think that you need to state that it is optional, as you can see it in the signature (by having a default value).
The signature should be the single point of truth.